src/utils.ts

```diff
@@ -0,0 +1,33 @@
+import fs from 'fs/promises'
+import zlib from 'zlib'
+
+/**
+ * Opens a WASM file provided as `path`, compresses the file using
+ * GZIP compression and retrieves the compressed file `Buffer`.
+ *
+ *  @param path The path of the file in the fs
+ */
+export async function openWasmFile(path: string): Promise<Buffer> {
+    const file = await fs.readFile(path)
+    const compressed = await compressGzip(file)
+
+    return compressed
+}
+
+/**
+ * Compresses the provided `Buffer` using GZIP compression and retrieves a
+ * `Buffer` containing the compressed bytes.
+ *
+ *  @param buff The `Buffer` instance to compress
+ */
+export function compressGzip(buff: Buffer): Promise<Buffer> {
+    return new Promise((resolve, reject) => {
+        zlib.gzip(buff, (error, compressed) => {
+            if (error) {
+                return reject(error)
+            }
+
+            resolve(compressed)
+        })
+    })
+}
```

Comment:
Given that `gzip` uses a callback, I wrap the function call into a Promise in order to have `async`/`await` syntax support when using this function as a consumer.